File: elixir/bob/lib/bob.ex
defmodule Bob do
  def hey(input) do
    input = String.trim(input)

    cond do
      yelling?(input) and question?(input) -> "Calm down, I know what I'm doing!"
      yelling?(input) -> "Whoa, chill out!"
      question?(input) -> "Sure."
      silent?(input) -> "Fine. Be that way!"
      true -> "Whatever."
    end
  end

  def yelling?(input) do
    not silent?(input) and input == String.upcase(input) and input != String.downcase (input)
  end

  def question?(input) do
    String.last(input) == "?"
  end

  def silent?(input) do
    input == ""
  end
end

File: clojure/bob/src/bob.clj
(ns bob
  (:require [clojure.string :as str]))

(defn question? [s]
  (str/ends-with? s "?"))

(defn yelling? [s]
  (and (re-seq #"[a-zA-Z]" s) (= (str/upper-case s) s)))

(defn response-for [s] ;; <- arglist goes here
  (let [s (str/trim s)]
    (cond
          (and (question? s) (yelling? s)) "Calm down, I know what I'm doing!"
          (question? s) "Sure."
          (yelling? s) "Whoa, chill out!"
          (str/blank? s) "Fine. Be that way!"
          :else "Whatever.")))